execute.py

```diff
@@ -0,0 +1,759 @@
+
+        # get the ligand mm forces
+        ligand_state = self.ligand_context.getState(getForces=True)
+        ligand_mm_force_matrix = ligand_state.getForces(asNumpy=True)
+
+        # get the ligand ani forces
+        ligand_state = self.ligand_context.getState(getPositions=True)
+        coords = ligand_state.getPositions(asNumpy=True)
+        ligand_ani_force_matrix, energie = self.ani_handler.calculate_force(coords)
+
+        # now combine the ligand forces
+        ligand_force_matrix = self.lambda_sequence[lambda_index] * (ligand_ani_force_matrix - ligand_mm_force_matrix)
+
+        # and append to the complex forces...
+        complex_mm_force_matrix[self.ligand_subset_indices, :] += ligand_force_matrix
+
+        complex_force_matrix = complex_mm_force_matrix
+        return complex_force_matrix
+
+    def _create_sampler_states(self, complex_positions):
+        """
+        simple utility function to generate complex and ligand sampler states
+        """
+        from openmmtools.states import SamplerState
+        complex_sampler_state = SamplerState(positions=complex_positions,
+                                             box_vectors=self.complex_system.getDefaultPeriodicBoxVectors())
+        ligand_positions = unit.Quantity(np.zeros([len(self.ligand_subset_indices), 3]), unit=unit.nanometer)
+        ligand_positions[self.ligand_subset_indices, :] = complex_sampler_state.positions[self.ligand_subset_indices, :]
+        ligand_sampler_state = SamplerState(positions=ligand_positions,
+                                            box_vectors=self.ligand_system.getDefaultPeriodicBoxVectors())
+
+        return complex_sampler_state, ligand_sampler_state
+
+    def simulate_baoab(self,
+                       _lambda_index,
+                       n_steps=1,
+                       gamma=1. / unit.picoseconds,
+                       dt=1.0 * unit.femtoseconds,
+                       metropolize=False):
+        """
+        Simulate n_steps of BAOAB, accumulating heat
+        
+        args
+            n_steps : int, default 1
+                number of steps of dynamics to run
+            gamma : unit.quantity.Quantity (units compatible with 1.0/unit.picoseconds)
+                collision rate
+            dt : unit.quantity.Quantity (units compatible with units.femtoseconds)
+                timestep
+        """
+        Q = 0
+        W_shads = 0.
+        _complex_sampler_state, _ligand_sampler_state = SamplerState.from_context(
+            self.complex_context), SamplerState.from_context(self.ligand_context)
+        x_unit, v_unit, t_unit, m_unit = unit.nanometers, unit.nanometers / unit.femtoseconds, unit.femtoseconds, unit.daltons
+        x, v = _complex_sampler_state.positions.value_in_unit(
+            unit.nanometers), _complex_sampler_state.velocities.value_in_unit(unit.nanometers / unit.femtoseconds)
+        dt = dt.value_in_unit(t_unit)
+        gamma = gamma * t_unit
```

Comment:
What does this line do? (gamma has unit 1/t_unit)

Reply:
make gamma unitless?

Reply:
Okay!

The gamma-dependent term that appears in an exponential (`gamma * dt`) will already be unitless even when gamma and dt are unit'd.

(Throughout, I think units are being removed a bit "preemptively", not sure I follow the intent of this.)